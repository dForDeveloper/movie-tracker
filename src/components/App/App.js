import React, { Component } from 'react';
import { fetchData } from '../../utils/api';

class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount = async () => {
    console.log(await fetchData('https://api.themoviedb.org/discover/movie?primary_release_year=2018'));
  } 

  render() {
    return (
      <div className="App">
      </div>
    );
  }
}

export default App;
