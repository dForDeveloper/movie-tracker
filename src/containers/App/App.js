import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchMovies } from '../../thunks/fetchMovies';
import LoginForm from '../LoginForm/LoginForm';
import MovieContainer from '../MovieContainer/MovieContainer';
import MovieDetails from '../../components/MovieDetails/MovieDetails';
import NavBar from '../../containers/NavBar/NavBar';
import SignUpForm from '../../containers/SignUpForm/SignUpForm';

export class App extends Component {
  componentDidMount = () => {
    const apiKey = process.env.REACT_APP_API_KEY;
    const url = 
      `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}`;
    this.props.fetchMovies(url);
  }
  
  render() {
    const { movies } = this.props;
    return (
      <div className="App">
        <header className="title-section">
            <Link to='/'><h1 className="app-title">Movie Tracker</h1></Link>
            <h2 className="app-subtitle">Find and save your favorite new releases.</h2>
        </header>
        <Route path='/' component={NavBar} />
        <Route path='/sign-up' component={SignUpForm} />
        <Route path='/login' component={LoginForm} />
        <Route path='/favorites' component={MovieContainer}/>
        <Route exact path='/' component={MovieContainer} />
        <Route path='/movies/:id' render={({ match }) => {
          const { id } = match.params;
          const movie = movies.find(movie => movie.id === parseInt(id));
          return movie ? <MovieDetails {...movie} /> : null;
        }} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  fetchMovies: (url) => dispatch(fetchMovies(url))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));

App.propTypes = {
  movies: PropTypes.array,
  addMovies: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object
};