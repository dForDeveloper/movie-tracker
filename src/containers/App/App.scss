.App {
  font-family: $body-font;
  color: $body-font;
}

.app-title {
  text-align: center;
  font-family: $heading-font;
  font-size: 60px;
  margin: 0px;
  padding: 20px;
  background-image: linear-gradient(#8F9DC3, #D3CEED);
  text-shadow: 10px 10px black;
}

a {
  text-decoration: none;
  color: #164946;
}

a:active {
  color: $steel-grey;
}

html {
  height: 100%;
  width: 100%;
}

body {
  background-image: url("images/background.jpg");
  background-attachment: fixed;
  background-repeat: no-repeat;
  background-size: cover;
}