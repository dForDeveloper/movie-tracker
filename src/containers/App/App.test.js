import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });

  describe('App container', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {});

  describe('mapDispatchToProps', () => {});
  
});
