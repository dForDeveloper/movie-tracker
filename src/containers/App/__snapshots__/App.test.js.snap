// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App container should properly render the component elements 1`] = `
<div
  className="App"
>
  <Link
    replace={false}
    to="/"
  >
    <h1
      className="h1"
    >
      Movie Tracker
    </h1>
  </Link>
  <Route
    component={[Function]}
    path="/sign-up"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Route
    component={[Function]}
    path="/favorites"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    path="/movies/:id"
    render={[Function]}
  />
</div>
`;
